rules/prefer-dom-node-dataset.js

```diff
@@ -46,6 +46,7 @@ const create = context => ({
 			}
 
 			case 'hasAttribute':
+				// TODO: Use `Object.hasOwn()` here when targeting Node.js 16.
```

Comment:
Arr you sure you want this comment? This code should running in browser.

Reply:
`Object.hasOwn()` is already in Chrome, Safari, and Firefox.